packages/material-ui-lab/src/TreeNode/TreeNode.test.js

```diff
@@ -0,0 +1,555 @@
+import React from 'react';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { createMount, getClasses } from '@material-ui/core/test-utils';
+import describeConformance from '@material-ui/core/test-utils/describeConformance';
+import { cleanup, createClientRender, fireEvent } from 'test/utils/createClientRender';
+import TreeNode from './TreeNode';
+import TreeView from '../TreeView';
+
+describe('<TreeNode />', () => {
+  const render = createClientRender({ strict: false });
+  const mount = createMount({ strict: false });
+  const classes = getClasses(<TreeNode nodeId="one" label="one" />);
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describeConformance(<TreeNode nodeId="one" label="one" />, () => ({
+    classes,
+    inheritComponent: 'li',
+    mount,
+    refInstanceof: window.HTMLLIElement,
+    skip: ['componentProp'],
+    after: () => mount.cleanUp(),
+  }));
+
+  it('should call onClick when clicked', () => {
+    const handleClick = spy();
+
+    const { getByText } = render(
+      <TreeView>
+        <TreeNode nodeId="test" label="test" onClick={handleClick} />
+      </TreeView>,
+    );
+
+    fireEvent.click(getByText('test'));
+
+    expect(handleClick.callCount).to.equal(1);
+  });
+
+  it('should call onFocus when focused', () => {
+    const handleFocus = spy();
+
+    const { getByText } = render(
+      <TreeView>
+        <TreeNode nodeId="test" label="test" onFocus={handleFocus} />
+      </TreeView>,
+    );
+
+    fireEvent.focus(getByText('test'));
+
+    expect(handleFocus.callCount).to.equal(2);
+  });
+
+  it('should call onKeyDown when clicked', () => {
+    const handleKeyDown = spy();
+
+    const { getByText } = render(
+      <TreeView>
+        <TreeNode nodeId="test" label="test" onKeyDown={handleKeyDown} />
+      </TreeView>,
+    );
+
+    fireEvent.keyDown(getByText('test'), { key: 'Enter' });
+
+    expect(handleKeyDown.callCount).to.equal(1);
+  });
+
+  describe('Accessibility', () => {
+    it('should have the role `treeitem`', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test" />
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[role="treeitem"]')).to.be.ok;
+    });
+
+    it('should add the role `group` to a `ul` component if it has children', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test">
+            <TreeNode nodeId="test2" label="test" />
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('ul[role="group"]')).to.be.ok;
+    });
+
+    it('should have the attribute `aria-expanded=false` if collapsed', () => {
```

Comment:
What should have this attribute? This is testing that an arbitrary element has this attribute. You probably meant to test something like `expect(getByRole(whatRole?)).not.to.have.attribute('aria-expanded', 'true')`.